modules/uncertainty/R/generate_joint_ensemble_design.R

```diff
@@ -51,29 +52,34 @@ generate_joint_ensemble_design <- function(settings,
     sampled_inputs[[input_tag]] <- input_result$ids
     design_list[[input_tag]] <- input_result$ids
   }
-
-  # Sample parameters
-  PEcAn.uncertainty::get.parameter.samples(
-    settings,
-    ensemble.size = ensemble_size,
-    posterior.files,
-    ens.sample.method
-  )
-
+  
+  # Sample parameters if we don't have it.
+  if (is.null(ensemble_samples)) {
+    PEcAn.uncertainty::get.parameter.samples(
+      settings,
+      ensemble.size = ensemble_size,
+      posterior.files,
+      ens.sample.method
+    )
+    samples.file <- file.path(settings$outdir, "samples.Rdata")
+  }
+  
   # Load samples from file
-  samples.file <- file.path(settings$outdir, "samples.Rdata")
   samples <- new.env()
-  if (file.exists(samples.file)) {
-    load(samples.file, envir = samples)
-    if (!is.null(samples$ensemble.samples)) {
-      # Just a placeholder: extract representative trait index per ensemble member
-      # You may want to flatten or select indices per trait
-      design_list[["param"]] <- seq_len(ensemble_size)
+  # if we don't have the parameters from the outside. 
+  if (is.null(ensemble_samples)) {
+    if (ile.exists(samples.file)) {
+      load(samples.file, envir = samples)
     } else {
-      PEcAn.logger::logger.warn("ensemble.samples not found in samples.Rdata")
+      PEcAn.logger::logger.error(samples.file, "not found, this file is required")
     }
+  }
+  if (!is.null(samples$ensemble.samples) | !is.null(ensemble_samples)) {
+    # Just a placeholder: extract representative trait index per ensemble member
+    # You may want to flatten or select indices per trait
+    design_list[["param"]] <- seq_len(ensemble_size)
   } else {
-    PEcAn.logger::logger.error(samples.file, "not found, this file is required")
+    PEcAn.logger::logger.warn("ensemble.samples not found in samples.Rdata")
   }
```

Comment:
Add `if-control` for the case where we don't assign the `ensemble_samples` argument.